splice_machine/src/main/java/com/splicemachine/derby/impl/sql/compile/MergeJoinStrategy.java

```diff
@@ -83,6 +94,15 @@ public boolean feasible(Optimizable innerTable,
         if(JoinStrategyUtil.isNonCoveringIndex(innerTable)){
             return false;
         }
+
+        // Merge joins involving the target of an UPDATE or DELETE can be slow on Spark
+        // because the input splits for a table which is getting frequently updated
+        // are currently not calculated accurately, leading to uneven splits.
+        // It is safer to pick a join which will repartition the tables into multiple even
+        // partitions such as MergeSortJoin.
+        if (optimizer.isForSpark() && !wasHinted && joinHasTargetTable(innerTable, optimizer))
+            return false;
+
```

Comment:
I think the bug here is that we are not computing good splits. Can you reference https://splicemachine.atlassian.net/browse/DB-7642 here ? Once that's fixed we should remove this restriction.

Reply:
Added a TODO comment referencing DB-7642